kernel/src/ipc.rs

```diff
@@ -187,33 +206,104 @@ impl<const NUM_PROCS: usize> LegacyDriver for IPC<NUM_PROCS> {
     /// Returns EINVAL if the other process doesn't exist.
     fn command(
         &self,
+        command_number: usize,
         target_id: usize,
-        client_or_svc: usize,
         _: usize,
         appid: AppId,
-    ) -> ReturnCode {
-        let cb_type = if client_or_svc == 0 {
-            IPCCallbackType::Service
-        } else {
-            IPCCallbackType::Client
-        };
+    ) -> CommandResult {
+        match command_number {
+            0 => {
+                self.data
+                    .enter(appid, |data, _| {
+                        data.search_slice.map_or(
+                            CommandResult::failure(ErrorCode::INVAL),
+                            |slice| {
+                                self.data
+                                    .kernel
+                                    .process_until(|p| {
+                                        let s = p.get_process_name().as_bytes();
+                                        // are slices equal?
+                                        if s.len() == slice.len()
+                                            && s.iter().zip(slice.iter()).all(|(c1, c2)| c1 == c2)
+                                        {
+                                            Some(CommandResult::success_u32(p.appid().id() as u32))
```

Comment:
Based on your description in this PR, and how this used to work when it was part of `allow()`, shouldn't this be `p.appid().id() + 1`?

Reply:
Yes! Missed that thanks